src/api-service/__app__/onefuzzlib/orm.py

```diff
@@ -249,6 +249,27 @@ def hide_secrets(data: B, hider: Callable[[SecretData], SecretData]) -> B:
     return data
 
 
+def delete_secrets(data: B, deleter: Callable[[SecretData], None]) -> None:
```

Comment:
This doesn't actually delete unless the callback does deletion, right?

Maybe this should just be:
```suggestion
def visit_secrets(data: B, visitor: Callable[[SecretData], None]) -> None:
```

Then the meaning is clear at the use sites, which might look like:
```python
visit_secrets(self, delete_remote_secret_data)
```

Reply:
Since we're not renaming this, I suggest just adding a function/doc comment that emphasizes that is is the responsibility of the `deleter` callback to perform any actual deletion.